lib/message_bus/backends/redis.rb

```diff
@@ -253,9 +253,15 @@ def subscribe(channel, last_id = nil)
       def global_unsubscribe
         if @redis_global
           # new connection to avoid deadlock
-          new_redis_connection.publish(redis_channel_name, UNSUB_MESSAGE)
-          @redis_global.disconnect
-          @redis_global = nil
+          begin
+            new_redis = new_redis_connection
+            new_redis.publish(redis_channel_name, UNSUB_MESSAGE)
+            @redis_global.disconnect
+            @redis_global.disconnect!
```

Comment:
this double disconnect is confusing, why do we need it?

Reply:
`@redis_global.disconnect` disconnects from the Redis sub. While `@redis_global.disconnect!` actually disconnects the client.

Reply:
won't disconnect! do both?

Reply:
That is a good point. I guess we just want to close the connection here.

Reply:
Actually we can't, we need to `disconnect` from the sub first because calling `disconnect!` will cause an error to be thrown.

Reply:
yikes, that feels like a redis gem bug to me... disconnect! is meant to be rude and error less. Maybe do a PR to redis gem and then link to it from the source?

Reply:
O actually `@redis_global.disconnect` is meant to wait until we unsubscribe from the subscription here. The error thrown if we straight up `disconnect!` is because our `on.unsubscribe` callback which calls `@redis_global.unsubscribe` will be working with a `nil` object.

Reply:
maybe we fix that? then we don't need a double disconnect?